codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java

```diff
@@ -227,6 +228,11 @@ public static void headerListSizeExceeded(int streamId, long maxHeaderListSize)
         throw streamError(streamId, PROTOCOL_ERROR, "Header size exceeded max allowed size (%d)", maxHeaderListSize);
     }
 
+    public static void headerListSizeExceededOnDecode(int streamId, long maxHeaderListSize) throws Http2Exception {
```

Comment:
do we have to differentiate `onDecode`? Maybe just rename `DecodeHeaderListSizeException` to `HeaderListSizeException` and add a `boolean duringDecode()` method if we need to distinguish the inbound/outbound characteristics.